ansible/roles/ci/code_style_check/common_resources/files/files_by_extension.py

```diff
@@ -37,29 +39,79 @@ def argparser():
 
 
 def gather_all_python_files(file_path):
+    """This function walks through the given path collects all python files (with and without extension based on
+       header). It will exclude all python files found in the lint ignore file."""
     output = []
     for dirpath, _, filenames in os.walk(file_path):
+        excluded_files = gather_excluded_files(dirpath, "python")  # Gathers all the files to be ignored in this folder
         for filename in filenames:
-            file_path = f"{dirpath}/{filename}"
-            try:
-                fline = ""
-                with open(file_path) as python_file:
-                    fline = python_file.readline().strip()
-                if any(fline in head for head in PYTHON_HEADERS) and fline != "" and fline != "#":
-                    output.append(file_path)
-            except Exception:  # We don't actually care about the exception as it wont fail on python files.
+            full_file_path = os.path.join(dirpath, filename)
+            valid_file = check_for_python_file(full_file_path)
+            if filename in excluded_files:  # Skip excluded files
                 continue
+            if valid_file:
+                output.append(full_file_path)
     return output
 
 
 def gather_all_cpp_files(file_path):
+    """This function walks through the given path collects all c, cpp, h, hpp files It will exclude all python files
+       found in the lint ignore file."""
     output = []
     for dirpath, _, filenames in os.walk(file_path):
+        excluded_files = gather_excluded_files(dirpath, "cpp")  # Gathers all the files to be ignored in this folder
         for filename in filenames:
+            if filename in excluded_files:  # Skip excluded files
+                continue
             if os.path.splitext(filename)[1] in CPP_HEADERS:
-                output.append(f"{dirpath}/{filename}")
+                output.append(os.path.join(dirpath, filename))
     return output
 
+def gather_excluded_files(folder_path, filetype):
+    """Gatheres all of the files to ignore in the lint ignore file (if it exists). A * tells it to skip all files
+       or you can list indiviual files like `exclude_files=test.py,hello.cpp,hi.h`."""
```

Comment:
Can you make the docstring clearer? Where should the asterisk be defined - in the filetype, or LINT_IGNORE_FILE (it's obvious if you read the code but on doc level, it's not so clear).